geowebcache/kml/src/main/java/org/geowebcache/service/kml/KMLService.java

```diff
@@ -235,6 +239,14 @@ public void handleRequest(Conveyor conv) throws GeoWebCacheException {
             }
         }
         tile.setTileLayer(layer);
+        
+        if(Objects.nonNull(tile.getLayer())) {
+            if(tile.getTileIndex()[2]<0) {
```

Comment:
So, what does this mean? Either a comment here or some javadocs on ConveyorTile.getTileIndex() would be nice.

From what I can tell, looks like it might be checking the zoom level?

Reply:
The KML request parser uses negative zoom levels as magic numbers.  I'll add a comment.